src/huggingface_hub/hf_api.py

```diff
@@ -166,10 +166,26 @@ class RepoFile:
             file name, relative to the repo root. This is the only attribute
             that's guaranteed to be here, but under certain conditions there can
             certain other stuff.
+        size (`int`, *optional*):
+            The file's size, in bytes. This attribute is present when `files_metadata` argument
+            of [`repo_info`] is set to `True`. It's `None` otherwise.
+        blob_id (`str`, *optional*):
+            The file's git OID. This attribute is present when `files_metadata` argument
+            of [`repo_info`] is set to `True`. It's `None` otherwise.
+        lfs (`dict`, *optional*):
+            The file's LFS metadata (has two keys: `sha256` and `size`). This attribute is present when
+            `files_metadata` argument of [`repo_info`] is set to `True` and the file is stored
+            with Git LFS. It's `None` otherwise.
     """
 
     def __init__(self, rfilename: str, **kwargs):
         self.rfilename = rfilename  # filename relative to the repo root
+
+        # Optional file metadata
+        self.size: Optional[int] = kwargs.pop("size", None)
+        self.blob_id: Optional[str] = kwargs.pop("blobId", None)
+        self.lfs: Optional[dict] = kwargs.pop("lfs", None)
```

Comment:
This means that these will always be present on this object even if we didn't ask for optional metadata. I think it would result in leaner objects if we just removed this and had the two lines below 

```
        for k, v in kwargs.items():
            setattr(self, k, v)
```

take care of the attribution. This way no `self.blob_id == None` on all objects when not asking for this info. 

WDYT?

Reply:
Honestly, I am not a fan of dynamic objects like this
While I understand it can be practical as a maintainer, I'm not convinced by the user experience for several reasons:
- It does not play well with type checkers
- Documentation basically says "there can be more stuff in here, but I can't tell you what'"
- On a personal note, I would prefer a dict-like interface (`__getitem__` vs `setattr`). IMO, `"key" in repo_file` is nicer (more idiomatic?) than `hasattr(repo_file, "key")`

TL;DR: I'd rather have `blob_id` always set but sometimes `None`, than having it set only from times to times

That being said, I can make the change if maintainers think the proposed `setattr` loop is better

Reply:
I understand where you're coming from, and I don't have a super strong feeling about this so if you find it more simple this way, happy to keep it like this.

Reply:
What do others think about this ? @osanseviero @adrinjalali @julien-c

Reply:
I also struggle both as a user and as a developer to understand what an object does when its `__init__` accepts everything and the possible attributes are not documented.

In a case like this, I would find an API which is very explicit on what the attributes can be much more understandable.

If we really need to have some optional extra parameters, we can have them in a separate attribute which is a dict and includes "other" stuff.

For example, the docstring of this class doesn't even mention that everything can be passed to its constructor and that they're stored.

Also, from what I see our `*Info` classes can be all dataclasses, can't they?

Reply:
Small pref to have these attributes all the time, but set to None by default.

> I also struggle both as a user and as a developer to understand what an object does when its __init__ accepts everything and the possible attributes are not documented.

I think it's fine if those if we add them to `__init__` with a default None value. It should make it clearer that those attributes exist all the time, and have a default None value.

> Also, from what I see our *Info classes can be all dataclasses, can't they?

I think dataclasses can't have **kwargs if I'm not mistaken

Reply:
Note that we already have the pattern of having attributes set to `None` by default in `huggingface_hub`, so this approach would be consistent with the rest

E.g.
```
list_models(limit=2)
```
The above will give `config:None` for example, while the following will set the config to the right dictionary.

```
list_models(limit=2, fetch_config=True)
```

Reply:
For consistency with e.g. `ModelInfo`, I'm would add them to `__init__` with a default None value.

The reason for the `setattr` catch-all in all those classes is to maximize forward compatibility: if we add something to the server (which we sometimes do), past versions of the library support it out of the box

Also related, note that in the past a few versions of `huggingface_hub` required a non-optional attribute that was deprecated (the infamous `key`, cc @lhoestq) and that . So it's important that everything is kept optional